### PR TITLE
【マークアップ】トップページ

### PR DESCRIPTION
**Pull request by NomuraIsao5656**
# what
 トップページの実装

#why
 最終課題達成のため

ヘッダー及びフッターは花原さんが作成していますので、完了後に反映させる予定です

### CODE REVIEW

File: app/assets/stylesheets/goods.scss [hunk 1]
```diff
@@ -1,3 +1,522 @@
...
+          line-height: 1.4;
+        }
+        .text{
+          font-size: 18px;
+          margin: 20px 0 30px;
+          line-height: 1.5;    
+        }
+        .btn{
+          display: flex;
+          .aBtn{
```
Comment (hanaharafumiki) [inline app/assets/stylesheets/goods.scss]: > __[stylelint]__ Expected selector ".wrapper .Main .mainMiddleDl .mainMiddleDlInner .btn .aBtn" to come before selector ".wrapper .Main .mainMainVisual .contennt .contentBtn .aBtn:hover" ([see on Sider](https://sider.review/gh/repos/239745641/pulls/8#issue-5207654))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `no-descending-specificity` | error |
>
> References:
> - <https://github.com/stylelint/stylelint/tree/13.0.0/lib/rules/no-descending-specificity>
>
> </details>

.wrapper .Main .mainMiddleDl .mainMiddleDlInner .btn .aBtnと、
.wrapper .Main .mainMainVisual .contennt .contentBtn .aBtn：hover の順序が逆のようです！

File: app/assets/stylesheets/goods.scss [hunk 2]
```diff
@@ -1,3 +1,522 @@
...
+            width: 180px;
+            margin-right: 10px;
+          }
+          .aBtn:hover{
+            opacity: 0.6;
+          }
+
+          .gBtn{
```
Comment (hanaharafumiki) [inline app/assets/stylesheets/goods.scss]: > __[stylelint]__ Expected selector ".wrapper .Main .mainMiddleDl .mainMiddleDlInner .btn .gBtn" to come before selector ".wrapper .Main .mainMainVisual .contennt .contentBtn .gBtn:hover" ([see on Sider](https://sider.review/gh/repos/239745641/pulls/8#issue-5207655))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `no-descending-specificity` | error |
>
> References:
> - <https://github.com/stylelint/stylelint/tree/13.0.0/lib/rules/no-descending-specificity>
>
> </details>

同じく hoverの順序が異なるようです

File: app/assets/stylesheets/goods.scss [hunk 3]
```diff
@@ -1,3 +1,522 @@
...
+                ul{
+                  display: flex;
+                  align-items: center;
+                  justify-content: space-between;
+                  li{
+                    margin: 0;
+                    padding: 0;
+                    .i{
+                      display: inline-block;
+                      font: normal normal normal 14px/1 FontAwesome;
```
Comment (hanaharafumiki) [inline app/assets/stylesheets/goods.scss]: > __[stylelint]__ Unexpected missing generic font family ([see on Sider](https://sider.review/gh/repos/239745641/pulls/8#issue-5207656))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `font-family-no-missing-generic-family-keyword` | error |
>
> References:
> - <https://github.com/stylelint/stylelint/tree/13.0.0/lib/rules/font-family-no-missing-generic-family-keyword>
>
> </details>

フォントの指定がおかしいようです...！

File: app/assets/stylesheets/goods.scss [hunk 4]
```diff
@@ -1,3 +1,522 @@
...
+                        font-style: normal;
+                        font-variant-ligatures: normal;
+                        font-variant-caps: normal;
+                        font-variant-numeric: normal;
+                        font-variant-east-asian: normal;
+                        font-weight: normal;
+                        font-stretch: normal;
+                        font-size: inherit;
+                        line-height: 1;
+                        font-family: FontAwesome;
```
Comment (hanaharafumiki) [inline app/assets/stylesheets/goods.scss]: > __[stylelint]__ Unexpected missing generic font family ([see on Sider](https://sider.review/gh/repos/239745641/pulls/8#issue-5207657))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `font-family-no-missing-generic-family-keyword` | error |
>
> References:
> - <https://github.com/stylelint/stylelint/tree/13.0.0/lib/rules/font-family-no-missing-generic-family-keyword>
>
> </details>

同じく、フォントの記載がおかしいようです

File: app/assets/stylesheets/goods.scss [hunk 5]
```diff
@@ -1,3 +1,522 @@
...
+                      font-size: inherit;
```
Comment (hanaharafumiki) [inline app/assets/stylesheets/goods.scss]: > __[stylelint]__ Unexpected duplicate "font-size" ([see on Sider](https://sider.review/gh/repos/239745641/pulls/8#issue-5207658))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `declaration-block-no-duplicate-properties` | error |
>
> References:
> - <https://github.com/stylelint/stylelint/tree/13.0.0/lib/rules/declaration-block-no-duplicate-properties>
>
> </details>

font-sizeが重複しています

File: app/assets/stylesheets/goods.scss [hunk 6]
```diff
@@ -1,3 +1,522 @@
...
+              }
+            }
+          }
+        }
+
+      }
+    }
+    .mainPicupBrand{
+      padding-bottom: 80px;
+      padding: 80px 0 0;
```
Comment (hanaharafumiki) [inline app/assets/stylesheets/goods.scss]: > __[stylelint]__ Unexpected shorthand "padding" after "padding-bottom" ([see on Sider](https://sider.review/gh/repos/239745641/pulls/8#issue-5207659))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `declaration-block-no-shorthand-property-overrides` | error |
>
> References:
> - <https://github.com/stylelint/stylelint/tree/13.0.0/lib/rules/declaration-block-no-shorthand-property-overrides>
>
> </details>

 padding-bottomとpaddingを1行にまとめましょう！

File: app/assets/stylesheets/goods.scss [hunk 7]
```diff
@@ -1,3 +1,522 @@
...
+          font-size: 35px;
+          text-shadow: 0 0 5px rgba(0,0,0,0.4);
+        }
+      }
+      .rowDlBtn{
+        width: 100%;
+        display: flex;
+        justify-content: center;
+        margin: 20px auto 0;
+        .aBtn{
```
Comment (hanaharafumiki) [inline app/assets/stylesheets/goods.scss]: > __[stylelint]__ Expected selector ".wrapper .Main .mainRowDl .rowDlBtn .aBtn" to come before selector ".wrapper .Main .mainMainVisual .contennt .contentBtn .aBtn" ([see on Sider](https://sider.review/gh/repos/239745641/pulls/8#issue-5207660))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `no-descending-specificity` | error |
>
> References:
> - <https://github.com/stylelint/stylelint/tree/13.0.0/lib/rules/no-descending-specificity>
>
> </details>

ネストの順序が異なる様です！

Comment (hanaharafumiki) [inline app/assets/stylesheets/goods.scss]: > __[stylelint]__ Expected selector ".wrapper .Main .mainRowDl .rowDlBtn .aBtn" to come before selector ".wrapper .Main .mainMainVisual .contennt .contentBtn .aBtn:hover" ([see on Sider](https://sider.review/gh/repos/239745641/pulls/8#issue-5207661))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `no-descending-specificity` | error |
>
> References:
> - <https://github.com/stylelint/stylelint/tree/13.0.0/lib/rules/no-descending-specificity>
>
> </details>

同じく、ネストの順序が異なる様です！

Comment (hanaharafumiki) [inline app/assets/stylesheets/goods.scss]: > __[stylelint]__ Expected selector ".wrapper .Main .mainRowDl .rowDlBtn .aBtn" to come before selector ".wrapper .Main .mainMiddleDl .mainMiddleDlInner .btn .aBtn" ([see on Sider](https://sider.review/gh/repos/239745641/pulls/8#issue-5207662))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `no-descending-specificity` | error |
>
> References:
> - <https://github.com/stylelint/stylelint/tree/13.0.0/lib/rules/no-descending-specificity>
>
> </details>

こちらもネストの順序

Comment (hanaharafumiki) [inline app/assets/stylesheets/goods.scss]: > __[stylelint]__ Expected selector ".wrapper .Main .mainRowDl .rowDlBtn .aBtn" to come before selector ".wrapper .Main .mainMiddleDl .mainMiddleDlInner .btn .aBtn:hover" ([see on Sider](https://sider.review/gh/repos/239745641/pulls/8#issue-5207663))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `no-descending-specificity` | error |
>
> References:
> - <https://github.com/stylelint/stylelint/tree/13.0.0/lib/rules/no-descending-specificity>
>
> </details>

順序を...！

File: app/assets/stylesheets/goods.scss [hunk 8]
```diff
@@ -1,3 +1,522 @@
...
+          width: 180px;
+          margin-right: 10px;
+        }
+        .aBtn:hover{
```
Comment (hanaharafumiki) [inline app/assets/stylesheets/goods.scss]: > __[stylelint]__ Expected selector ".wrapper .Main .mainRowDl .rowDlBtn .aBtn:hover" to come before selector ".wrapper .Main .mainMainVisual .contennt .contentBtn .aBtn:hover" ([see on Sider](https://sider.review/gh/repos/239745641/pulls/8#issue-5207664))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `no-descending-specificity` | error |
>
> References:
> - <https://github.com/stylelint/stylelint/tree/13.0.0/lib/rules/no-descending-specificity>
>
> </details>

順序を...！

Comment (hanaharafumiki) [inline app/assets/stylesheets/goods.scss]: > __[stylelint]__ Expected selector ".wrapper .Main .mainRowDl .rowDlBtn .aBtn:hover" to come before selector ".wrapper .Main .mainMiddleDl .mainMiddleDlInner .btn .aBtn:hover" ([see on Sider](https://sider.review/gh/repos/239745641/pulls/8#issue-5207665))
>
> <details>
>
> | Rule | Severity |
> | - | - |
> | `no-descending-specificity` | error |
>
> References:
> - <https://github.com/stylelint/stylelint/tree/13.0.0/lib/rules/no-descending-specificity>
>
> </details>

順序を...！

File: app/views/goods/index.html.haml [hunk 9]
```diff
@@ -1 +1,214 @@
...
+            様々な支払いに対応
+          .itemDetail
+            お支払いは、クレジットカードだけでなく、ポイントや売上金など多彩な方法があります。
+
+
+    %section.mainPicupCategory
+      %h2.head ピックアップカテゴリー
+      .productBox
+        .productHead
+          %a{:href => "#"}
```
Comment (hanaharafumiki) [inline app/views/goods/index.html.haml]: =>
この ハッシュロケットは推奨されていないかもしれません...！
シンボル型に修正お願いします

File: app/views/goods/index.html.haml [hunk 10]
```diff
@@ -1 +1,214 @@
-テスト
\ No newline at end of file
+%head
+<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.3/css/all.css">
+
+.wrapper
```
Comment (yonmi08) [inline app/views/goods/index.html.haml]: 全て小文字から始まっています。
花原さんが送ってくださった、参考URLの確認をもう一度お願いします。
https://master.tech-camp.in/curriculums/2769

File: app/assets/stylesheets/goods.scss (hunk 3 above)
Comment (NomuraIsao5656) [reply app/assets/stylesheets/goods.scss]: 余分なフォント指定を消しました

File: app/assets/stylesheets/goods.scss (hunk 4 above)
Comment (NomuraIsao5656) [reply app/assets/stylesheets/goods.scss]: 余分なフォント指定を消しました

File: app/views/goods/index.html.haml (hunk 9 above)
Comment (NomuraIsao5656) [reply app/views/goods/index.html.haml]: href文をlink_to に直しました

File: app/views/goods/index.html.haml (hunk 10 above)
Comment (NomuraIsao5656) [reply app/views/goods/index.html.haml]: 大文字に直せ直せるものは修正しました。

**Comment by miyagit** [comment]
コンフリクトを直しましょう。

**Comment by NomuraIsao5656** [comment]
> コンフリクトを直しましょう。

修正しました

**Comment by TOMARUMARU** [comment]
まだconflictしているので解消しましょう
Siderも落ちているようなのでエラー解消しましょう

**Comment by TOMARUMARU** [comment]
LGTM

Sider ↓ だけ解消してmergeしましょう

**Pull request merged**